hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/ReliableTopicProxy.java

```diff
@@ -0,0 +1,245 @@
+
+    private final NodeEngine nodeEngine;
+    private final Address thisAddress;
+    private final String name;
+
+    public ReliableTopicProxy(String name, NodeEngine nodeEngine, ReliableTopicService service,
+                              ReliableTopicConfig topicConfig) {
+        super(nodeEngine, service);
+
+        this.name = name;
+        this.topicConfig = topicConfig;
+        this.nodeEngine = nodeEngine;
+        this.ringbuffer = nodeEngine.getHazelcastInstance().getRingbuffer(TOPIC_RB_PREFIX + name);
+        this.executor = initExecutor(nodeEngine, topicConfig);
+        this.thisAddress = nodeEngine.getThisAddress();
+        this.overloadPolicy = topicConfig.getTopicOverloadPolicy();
+
+        for (ListenerConfig listenerConfig : topicConfig.getMessageListenerConfigs()) {
+            addMessageListener(listenerConfig);
+        }
+    }
+
+    @Override
+    public String getServiceName() {
+        return ReliableTopicService.SERVICE_NAME;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    private void addMessageListener(ListenerConfig listenerConfig) {
+        NodeEngine nodeEngine = getNodeEngine();
+
+        MessageListener listener = loadListener(listenerConfig);
+
+        if (listener == null) {
+            return;
+        }
+
+        if (listener instanceof HazelcastInstanceAware) {
+            HazelcastInstanceAware hazelcastInstanceAware = (HazelcastInstanceAware) listener;
+            hazelcastInstanceAware.setHazelcastInstance(nodeEngine.getHazelcastInstance());
+        }
+        addMessageListener(listener);
+    }
+
+    private MessageListener loadListener(ListenerConfig listenerConfig) {
+        try {
+            MessageListener listener = (MessageListener) listenerConfig.getImplementation();
+            if (listener != null) {
+                return listener;
+            }
+
+            if (listenerConfig.getClassName() != null) {
+                Object object = ClassLoaderUtil.newInstance(nodeEngine.getConfigClassLoader(), listenerConfig.getClassName());
+
+                if (!(object instanceof MessageListener)) {
+                    throw new HazelcastException("class '"
+                            + listenerConfig.getClassName() + "' is not an instance of "
+                            + MessageListener.class.getName());
+                }
+                listener = (MessageListener) object;
+            }
+            return listener;
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
+    private Executor initExecutor(NodeEngine nodeEngine, ReliableTopicConfig topicConfig) {
+        Executor executor = topicConfig.getExecutor();
+        if (executor == null) {
+            executor = nodeEngine.getExecutionService().getExecutor(ASYNC_EXECUTOR);
+        }
+        return executor;
+    }
+
+    @Override
+    public void publish(E payload) {
+       //todo: exception???
```

Comment:
Currently there is no documentation if null is possible on itopic.publish. Let me see if the reliable topic can deal with it.